packages/runtime/runtime-utils/src/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -125,21 +132,21 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
      * - usedRoutes: This is used to figure out if the used state of this node changed since last summary.
      * - gcData: The garbage collection data of this node that is required for running GC.
      */
-    private async loadInitialGCSummaryDetails() {
-        if (this.gcDetailsInInitialSummaryP === undefined) {
+    private async loadBaseGCDetails() {
+        if (this.baseGCDetailsP === undefined) {
             return;
         }
 
-        const gcDetailsInInitialSummaryP = this.gcDetailsInInitialSummaryP;
-        this.gcDetailsInInitialSummaryP = undefined;
+        const baseGCDetailsP = this.baseGCDetailsP;
+        this.baseGCDetailsP = undefined;
```

Comment:
If there are two parallel calls to getBaseGCDetails(), then second call will return before data is loaded. Is that Ok?

Reply:
Even if it's Ok now, I'd suggest always waiting to avoid surprises in the future. I.e. first line should be 
await this.baseGCDetailsP, but rest of the logic should allow only one caller to do state permutations.

Reply:
Good catch. I had missed that the second call will return before data is loaded.